pyannote/audio/core/model.py

```
@@ -468,9 +477,8 @@ def __by_name(
         if isinstance(modules, str):
             modules = [modules]
 
-        for name, module in ModelSummary(self, max_depth=-1).named_modules:
-            if name not in modules:
-                continue
+        for name in modules:
+            module = getattr(self, name)
```

What happens if no module with this `name` exists?

---

This will raise an `AttributeError: <Model> object has no attribute <name>`.

---

we could raise an exception telling the user that there is no module named `name` for this model